### ISSUE TITLE
Accessibility fails when the  Popover is active

### ISSUE THREAD

**Issue by c9qsm62x** [Royal-Navy/design-system#3260]
### Steps to reproduce
1. Go to the Popover Click to activate demo https://storybook.design-system.digital.mod.uk/?path=/story/popover--click-to-activate
2. Click the "Click on me" button
3. Click another story then go back on to unmount and remount the component.
https://storybook.design-system.digital.mod.uk/?path=/story/popover--click-to-activate
### Minimal reproduction
![image](https://user-images.githubusercontent.com/75253719/165545418-2e388e1e-ca0e-4592-a501-2911696c2fcc.png)

**Comment by m7kvqbe1** [comment]
We'll action this soon.

This ticket shines light on an interesting problem - there are actually loads of intermediary states of components that probably aren't currently covered by the AXE tests. 

This is also an issue with Chromatic snapshots. There is an existing ticket in place to attempt to extend coverage.
